### PR TITLE
Feature: Have `apko_build` take the loaded `apko_config`

### PR DESCRIPTION
**Pull request by mattmoor**
:gift: This changes the type of `apko_build.config` to take the schematized APKO config.

Users can now pass a structured APKO configuration directly to `apko_build` or load it via `apko_config` and pass it through.

🚨  THIS IS A BREAKING CHANGE 🚨 

WIP until https://github.com/chainguard-dev/terraform-provider-apko/pull/51 merges

/kind feature

### CODE REVIEW

File: internal/provider/reflect.go [hunk 1]
```diff
@@ -136,6 +147,163 @@ func generateValue(v reflect.Value) (attr.Value, diag.Diagnostics) {
...
+	if rv.Kind() != reflect.Pointer && rv.Type().Name() != "" && rv.CanAddr() {
+		rv = rv.Addr()
+	}
+
+	return assignValueReflect(in, rv)
+}
+
+func assignValueReflect(in attr.Value, out reflect.Value) diag.Diagnostics {
+	t := out.Type()
+	// fmt.Printf("TYPE: %v %t\n", t, out.CanSet())
```
Comment (imjasonh) [inline internal/provider/reflect.go:166]:
```suggestion
```

File: internal/provider/reflect.go [hunk 2]
```diff
@@ -136,6 +147,163 @@ func generateValue(v reflect.Value) (attr.Value, diag.Diagnostics) {
...
+	case reflect.Slice:
+		lv, ok := in.(basetypes.ListValue)
+		if !ok {
+			return []diag.Diagnostic{diag.NewErrorDiagnostic("not a list", fmt.Sprintf("got: %T", in))}
+		}
+		elts := lv.Elements()
+
+		out = indirect(out)
+		t = out.Type()
+		// fmt.Printf("SLICE TYPE: %v %t\n", t, out.CanSet())
```
Comment (imjasonh) [inline internal/provider/reflect.go:221]:
```suggestion
```

File: internal/provider/reflect.go [hunk 3]
```diff
@@ -136,6 +147,163 @@ func generateValue(v reflect.Value) (attr.Value, diag.Diagnostics) {
...
+	case reflect.Map:
+		mv, ok := in.(basetypes.MapValue)
+		if !ok {
+			return []diag.Diagnostic{diag.NewErrorDiagnostic("not a map", fmt.Sprintf("got: %T", in))}
+		}
+		elts := mv.Elements()
+
+		out = indirect(out)
+		t = out.Type()
+		// fmt.Printf("MAP TYPE: %v %t\n", t, out.CanSet())
```
Comment (imjasonh) [inline internal/provider/reflect.go:251]:
```suggestion
```

File: internal/provider/reflect.go [hunk 4]
```diff
@@ -136,6 +147,163 @@ func generateValue(v reflect.Value) (attr.Value, diag.Diagnostics) {
...
+	case reflect.Struct:
+		ov, ok := in.(basetypes.ObjectValue)
+		if !ok {
+			return []diag.Diagnostic{diag.NewErrorDiagnostic("not a object", fmt.Sprintf("got: %T", in))}
+		}
+		fl := ov.Attributes()
+
+		out = indirect(out)
+		t = out.Type()
+		// fmt.Printf("STRUCT TYPE: %v %t\n", t, out.CanSet())
```
Comment (imjasonh) [inline internal/provider/reflect.go:277]:
```suggestion
```

**Comment by mattmoor** [comment]
I removed the debug print statements.  These were really helpful for debugging because the TF SDK does something to silence `log.Printf` even during tests.

**Pull request merged**